private/bufpkg/bufimage/bufimageutil/bufimageutil.go

```diff
@@ -737,15 +737,17 @@ func (t *transitiveClosure) exploreOptionValueForAny(
 	opts *imageFilterOptions,
 ) error {
 	switch {
-	case fd.IsMap() && isMessageKind(fd.MapValue().Kind()):
-		var err error
-		val.Map().Range(func(_ protoreflect.MapKey, v protoreflect.Value) bool {
-			if err = t.exploreOptionScalarValueForAny(v.Message(), referrerFile, imageIndex, opts); err != nil {
-				return false
-			}
-			return true
-		})
-		return err
+	case fd.IsMap():
```

Comment:
For map fields whose value was _not_ a map, it would previously fall-through and end up getting to line 760, where it would blow up with this:
```
type mismatch: cannot convert map to message
```
The value type is actually a map. But we end up calling the `Message()` method. Oops.

So now, if it's a map whose value is _not_ a message, it will no longer fall-through due to the condition change here.